packages/lib/src/treeUtils/draft.ts

```diff
@@ -0,0 +1,176 @@
+import { action, computed } from "mobx"
+import { pathToTargetPathIds } from "../actionMiddlewares/utils"
+import { resolvePath, resolvePathCheckingIds, skipIdChecking } from "../parent/path"
+import { Path } from "../parent/pathTypes"
+import { applyPatches } from "../patch/applyPatches"
+import { applySnapshot } from "../snapshot/applySnapshot"
+import { fromSnapshot } from "../snapshot/fromSnapshot"
+import { getSnapshot } from "../snapshot/getSnapshot"
+import { assertTweakedObject } from "../tweaker/core"
+import { failure } from "../utils"
+import { deepEquals } from "./deepEquals"
+
+/**
+ * A class that with draft related methods.
+ * Use `draft` to create an instance of this class.
+ *
+ * @typeparam T Data type.
+ */
+export class Draft<T extends object> {
+  /**
+   * Draft data object.
+   */
+  readonly data!: T
```

Comment:
Is `!` actually needed? `this.data` is set in the constructor.

Reply:
fixed